diff -urNp -x '*.orig' mailx-12.4.org/makeconfig mailx-12.4/makeconfig
--- mailx-12.4.org/makeconfig	2007-04-14 17:24:28.000000000 +0200
+++ mailx-12.4/makeconfig	2021-10-05 22:58:00.427635968 +0200
@@ -405,12 +405,10 @@ int main(void)
 }
 !
 
-<$tmp2.c link_check gssapi 'for GSSAPI in libgss' \
-		'#define USE_GSSAPI' '-lgss' ||
-	<$tmp2.c link_check gssapi 'for GSSAPI in libgssapi_krb5' \
-			'#define USE_GSSAPI' '-lgssapi_krb5' ||
-		link_check gssapi 'for GSSAPI in libgssapi_krb5, old-style' \
-				'#define USE_GSSAPI
+<$tmp2.c link_check gssapi 'for GSSAPI in libgssapi_krb5' \
+		'#define USE_GSSAPI' '-lgssapi_krb5' ||
+	link_check gssapi 'for GSSAPI in libgssapi_krb5, old-style' \
+			'#define USE_GSSAPI
 #define GSSAPI_OLD_STYLE' '-lgssapi_krb5' <<\! || \
 			link_check gssapi 'for GSSAPI in libgssapi' \
 				'#define USE_GSSAPI
