diff -dur mailx-8.1.1.orig/names.c mailx-8.1.1/names.c
--- mailx-8.1.1.orig/names.c	2004-01-27 11:32:23.405773493 +0100
+++ mailx-8.1.1/names.c	2004-01-27 11:32:53.980539061 +0100
@@ -51,6 +51,7 @@
 #include "rcv.h"
 #include <fcntl.h>
 #include "extern.h"
+#include <time.h>
 
 /*
  * Allocate a single element of a name list,
diff -dur mailx-8.1.1.orig/send.c mailx-8.1.1/send.c
--- mailx-8.1.1.orig/send.c	1996-06-14 10:27:08.000000000 +0200
+++ mailx-8.1.1/send.c	2004-01-27 11:33:11.505392024 +0100
@@ -44,6 +44,7 @@
 
 #include "rcv.h"
 #include "extern.h"
+#include <time.h>
 
 /*
  * Mail -- a mail program
@@ -489,19 +490,23 @@
 	FILE *fo;
 	int w;
 {
-	register int gotcha;
-
-	gotcha = 0;
+	time_t czas;
+	struct tm *czas_s;
+	char strdata[128];
+	
+	time(&czas);
+	czas_s = localtime(&czas);
+	strftime(strdata, 64, "%a, %d %b %Y %T %z", czas_s);
+	fprintf(fo, "Date: %s\n", strdata);
 	if (hp->h_to != NIL && w & GTO)
-		fmt("To:", hp->h_to, fo, w&GCOMMA), gotcha++;
+		fmt("To:", hp->h_to, fo, w&GCOMMA);
 	if (hp->h_subject != NOSTR && w & GSUBJECT)
-		fprintf(fo, "Subject: %s\n", hp->h_subject), gotcha++;
+		fprintf(fo, "Subject: %s\n", hp->h_subject);
 	if (hp->h_cc != NIL && w & GCC)
-		fmt("Cc:", hp->h_cc, fo, w&GCOMMA), gotcha++;
+		fmt("Cc:", hp->h_cc, fo, w&GCOMMA);
 	if (hp->h_bcc != NIL && w & GBCC)
-		fmt("Bcc:", hp->h_bcc, fo, w&GCOMMA), gotcha++;
-	if (gotcha && w & GNL)
-		(void) putc('\n', fo);
+		fmt("Bcc:", hp->h_bcc, fo, w&GCOMMA);
+	putc('\n', fo);
 	return(0);
 }
 
