--- mailx-8.1.1/extern.h~	2005-05-04 22:19:21.000000000 +0000
+++ mailx-8.1.1/extern.h	2005-05-04 22:20:55.000000000 +0000
@@ -163,7 +163,7 @@
 int	 isdir __P((char []));
 int	 isfileaddr __P((char *));
 int	 ishead __P((char []));
-int	 isign __P((char *, struct ignoretab []));
+int	 isign __P((char *, struct ignoretab *));
 int	 isprefix __P((char *, char *));
 void	 istrcpy __P((char *, char *, int));
 const struct cmd *
