diff -Naur mailx-8.1.1/pathnames.h mailx-8.1.1-p/pathnames.h
--- mailx-8.1.1/pathnames.h	Wed Oct  4 18:16:12 2000
+++ mailx-8.1.1-p/pathnames.h	Wed Oct  4 18:12:43 2000
@@ -37,6 +37,9 @@
  *	$NetBSD: pathnames.h,v 1.4 1996/06/08 19:48:34 christos Exp $
  */
 
+/* glibc defaults */
+#include <paths.h>
+ 
 /* mail installed files */
 #define _PATH_HELP	"/usr/share/misc/mail.help"
 #define _PATH_TILDE	"/usr/share/misc/mail.tildehelp"
