--- mailx-8.1.1.orig/USD.doc/Makefile
+++ mailx-8.1.1/USD.doc/Makefile
@@ -4,8 +4,10 @@
 SRCS=	mail0.nr mail1.nr mail2.nr mail3.nr mail4.nr mail5.nr mail6.nr \
 	mail7.nr mail8.nr mail9.nr maila.nr
 MACROS=	-me
+TBL=tbl
+manual.ps: ${SRCS}
+	${TBL} ${SRCS} | groff ${MACROS} -Tps > $@
 
-paper.ps: ${SRCS}
-	${TBL} ${SRCS} | ${ROFF} > ${.TARGET}
+clean :
+	-rm manual.ps
 
-.include <bsd.doc.mk>
--- mailx-8.1.1.orig/Makefile
+++ mailx-8.1.1/Makefile
@@ -2,22 +2,44 @@
 #	$NetBSD: Makefile,v 1.8 1996/06/08 19:48:09 christos Exp $
 
 PROG=	mail
+CC=gcc
+
+# use second line starting from hamm release
+#CPPFLAGS=-I/usr/include/bsd -D_BSD_SOURCE -DIOSAFE
+CPPFLAGS=-D_BSD_SOURCE 
+
+CFLAGS=-g
 SRCS=	version.c aux.c cmd1.c cmd2.c cmd3.c cmdtab.c collect.c dotlock.c \
 	edit.c fio.c getname.c head.c v7.local.c lex.c list.c main.c names.c \
 	popen.c quit.c send.c strings.c temp.c tty.c vars.c
+
+OBJS=$(SRCS:%.c=%.o)
+LIBS=
+
 SFILES=	mail.help mail.tildehelp
 EFILES=	mail.rc
 LINKS=	${BINDIR}/mail ${BINDIR}/Mail ${BINDIR}/mail ${BINDIR}/mailx
-MLINKS=	mail.1 Mail.1 mail.1 mailx.1
+MFILES=	mail.1
 
-beforeinstall:
-	cd ${.CURDIR}/misc; install -c -o ${BINOWN} -g ${BINGRP} \
-	    -m 444 ${SFILES} ${DESTDIR}/usr/share/misc
-	cd ${.CURDIR}/misc; install -c -o root -g wheel \
-	    -m 644 ${EFILES} ${DESTDIR}/etc
-
-.if make(install)
-SUBDIR+= USD.doc
-.endif
+default: all
 
-.include <bsd.prog.mk>
+ all: $(PROG)
+ 
+ $(PROG): $(OBJS)
+	$(CC) $(LDFLAGS) $(CFLAGS) -o $@ $(OBJS) $(LIBS)
+ 
+ .c.o:
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c $<
+ 
+ .y.c:
+	bison $<
+	mv -f $*.tab.c $@
+ 
+ clean:
+	rm -f $(PROG) *.o *~
+ 
+ install:
+	install -c -m 2755 -o root -g mail -s $(PROG) $(DESTDIR)/usr/bin/
+	install -c -m 644 $(MFILES) $(DESTDIR)/usr/man/man1/
+	cd misc && install -c -m 644 $(EFILES) $(DESTDIR)/etc/
+	cd misc && install -c -m 644 $(SFILES) $(DESTDIR)/usr/lib/
